### PR TITLE
feat: show warning when installing handheld on desktop

### PR DESCRIPTION
**Pull request by SoulHarsh007**
for handheld edition, use `chwd` to check if the device is a known handheld device or not. If it is not a known handheld device, show a warning that the user is trying to install the handheld edition on unsupported hardware

### CODE REVIEW

**Comment by SoulHarsh007** [comment]
Depends on: https://github.com/CachyOS/CachyOS-Live-ISO/pull/33 and https://github.com/CachyOS/CachyOS-Live-ISO/pull/34

File: i18n/de/cachyos_hello.ftl [hunk 1]
```diff
@@ -56,6 +56,7 @@ section-project = PROJEKT
 
 # Main Page (body)
 offline-error = Die Online-Installation kann nicht gestartet werden! Keine Internetverbindung
+unsupported-hw-warning = Sie versuchen, die Installation auf einer Hardware durchzuführen, die von der aktuellen ISO nicht unterstützt wird, und können daher keinen Support in Anspruch nehmen
```
Comment (vnepogodin) [inline i18n/de/cachyos_hello.ftl:59]: @ptr1337 help

File: i18n/pl/cachyos_hello.ftl [hunk 2]
```diff
@@ -56,6 +56,7 @@ section-project = PROJEKT
 
 # Main Page (body)
 offline-error = Instalacja online niemożliwa! Brak połączenia internetowego
+unsupported-hw-warning = Próba instalacji na sprzęcie nieobsługiwanym przez aktualne ISO spowoduje, że instalacja nie będzie objęta pomocą techniczną
```
Comment (vnepogodin) [inline i18n/pl/cachyos_hello.ftl:59]: @sirlucjan help

File: i18n/ru/cachyos_hello.ftl [hunk 3]
```diff
@@ -75,6 +75,7 @@ calamares-install-type = Calamares тип установки
 
 # Main Page (body)
 offline-error = Не удается запустить онлайн-установку! Нет подключения к Интернету
+unsupported-hw-warning = Вы пытаетесь установить на оборудование, не поддерживаемое текущим ISO, ваша установка не будет иметь права на поддержку
```
Comment (vnepogodin) [inline i18n/ru/cachyos_hello.ftl]: Sounds weird

File: src/main.rs [hunk 4]
```diff
@@ -41,6 +41,32 @@ static G_SAVE_JSON: Lazy<Mutex<serde_json::Value>> = Lazy::new(|| {
 });
 static mut G_HELLO_WINDOW: Option<Arc<HelloWindow>> = None;
 
+fn version_compat_check(message: String) {
+    let version_tag = fs::read_to_string("/etc/version-tag").unwrap_or_else(|_| "desktop".to_string());
```
Comment (vnepogodin) [inline src/main.rs]: Replace with /etc/edition-tag

File: src/main.rs [hunk 5]
```diff
@@ -41,6 +41,32 @@ static G_SAVE_JSON: Lazy<Mutex<serde_json::Value>> = Lazy::new(|| {
...
+
+    if version_tag == "handheld" {
+        let handheld_profiles = chwd::profile::parse_profiles(&format!("{}/handhelds/profiles.toml", chwd::consts::CHWD_PCI_CONFIG_DIR)).unwrap();
+        let handheld_profile_names: Vec<_> = handheld_profiles.iter().map(|x| &x.name).collect();
+
+        let mut exec_out = Exec::shell("chwd --list -d | grep Name | awk '{print $4}'")
```
Comment (ventureoo) [inline src/main.rs]: Can't we just expose some API functions in chwd lib for that?

File: src/main.rs [hunk 6]
```diff
@@ -41,6 +41,32 @@ static G_SAVE_JSON: Lazy<Mutex<serde_json::Value>> = Lazy::new(|| {
...
+            .stdout(Redirection::Pipe)
+            .capture()
+            .unwrap()
+            .stdout_str();
+        exec_out.pop();
+
+        if exec_out.split('\n').any(|profile| handheld_profile_names.contains(&&profile.to_owned())) {
```
Comment (vnepogodin) [inline src/main.rs]:
```suggestion
        let available_profiles = chwd::profile::get_available_profiles(false);
        if available_profiles.iter().any(|profile| handheld_profile_names.contains(profile.name)) {
```

File: src/main.rs [hunk 7]
```diff
@@ -41,6 +41,32 @@ static G_SAVE_JSON: Lazy<Mutex<serde_json::Value>> = Lazy::new(|| {
 });
 static mut G_HELLO_WINDOW: Option<Arc<HelloWindow>> = None;
 
+fn version_compat_check(message: String) {
+    let version_tag = fs::read_to_string("/etc/edition-tag").unwrap_or_else(|_| "desktop".to_string());
```
Comment (vnepogodin) [inline src/main.rs]:
```suggestion
    let version_tag = fs::read_to_string("/etc/edition-tag").unwrap_or( "desktop".to_string());
```

probably would be more sense to use unwrap_or here, we dont do any heavy operations for lazy eval

**Review by vnepogodin — CHANGES_REQUESTED** [review]

**Review by vnepogodin — APPROVED** [review]
Many thanks

**Pull request merged**